### ISSUE TITLE
[Question]: Not getting expected output when dynamic iteration inside filter for select query.

### ISSUE THREAD

**Issue by Rahulucky03** [supabase-community/supabase-kt#731]
### General info

- [X] I checked the [troubleshooting](https://github.com/supabase-community/supabase-kt/blob/master/TROUBLESHOOTING.md) page for similar problems

### What is your question?

I have a method to fetch all users from supabase db, inside the method one hashmap parameter to filter the users. when I apply dynamic filter its not getting expected results but when I hardcoded same filters in code directly Its working properly.
I am using Supabase-kt for my android application.

**Dynamic way :**
```
    suspend fun fetchUsers(
        filterMap: Map<String, List<String>> = mapOf()
    ): List<User> {
        val usersTable = postgrest.from(“users”)
        val column = Columns.raw(COLUMNS_WITH_MULTIPLE_TABLE_INCLUDING_JOINS)
        usersTable.select(column) {
            filter {
                //applying dynamic filter
                filterMap.map { (key, value) ->
                    if (value.isNotEmpty()) {
                        Log.d(TAG, value.toString())
                        filter(FilterOperation(key, FilterOperator.IN, "(${value.joinToString(",")})"))
                        //As we can use any of them (filter or isIn)
                        isIn(key, value)
                    }
                }
            }
        }
    }
```

**Static way:**
```
    suspend fun fetchUsers(
        //filterMap: Map<String, List<String>> = mapOf() //Not passing values
    ): List<User> {
        val usersTable = postgrest.from(“users”)
        val column = Columns.raw(COLUMNS_WITH_MULTIPLE_TABLE_INCLUDING_JOINS)
        usersTable.select(column) {
            filter {
                //applying static filter
                filter(FilterOperation(“tableName.columnName”, FilterOperator.IN, “(value1, value2)”))
                //As we can use any of them (filter or isIn)
                isIn(“tableName.columnName”, arrayListOf("value1", "value2"))
            }
        }
    }
```

In Dynamic way, not working. 
In static way getting expected results.

I cross confirm multiple times that I am making anything mistakes or not but I did not get anything rather this.

I have check below points as well:
- For the iteration of hashmap, I have used map and loop both (both not worked for dynamic scenario)
- Tried on version 2.0.0 and 2.2.2. I can't check other version due to many changes in the project. It can be risky as of now.
- I have double check the joins, table and column names.

Notes for Hashmap filters: 
- Key contains the exact string based on column name including table name if needed.
- Values contains the list of items in string form.

Can you help me with this issue?

**Comment by jan-tennert** [comment]
As you are using a rather old version, I cannot be certain, but can you print out the `params` within the filter builder after applying your dynamic filters?

**Comment by Rahulucky03** [comment]
Hey Team,

I am so sorry guys, I made a silly mistake, and because of that, it is not working, just 10 minutes ago. I solved it and now it is working very smoothly.

Actually, I worked on it for 5-6 days and I couldn't find the reason, even when I tried debug points and logs. But I couldn't figure it out. So I posted it here. Though I knew it was a simple scenario and it should work, but after spending so much time I only thought of posting it here.

I made a mistake while calling the fetchUsers function, This is calling from multiple places. so It's my mistake only. 

Supbase team is awesome :)
Thanks a lot, team for your support.
